numpy/lib/format.py

```diff
@@ -618,6 +621,15 @@ def write_array(fp, array, version=None, allow_pickle=True, pickle_kwargs=None):
         buffersize = max(16 * 1024 ** 2 // array.itemsize, 1)
 
     if array.dtype.hasobject:
+        if allow_pickle is None:
+            # TODO: remove in 1.18 after deprecation in 1.17
+            allow_pickle = True
+            msg = ("Object arrays will not be saved by default in the future"
+                    " because `allow_pickle` will default to False. You should"
+                    " add `allow_pickle=True` explicitly to eliminate this"
+                    " warning.")
+            warnings.warn(msg, FutureWarning, stacklevel=2)
+
```

Comment:
I don't think this is needed.

Reply:
In fact, emitting a warning here is bad since it can possibly raise an exception, which could cause the data to be thrown away, a frustrating experience when trying to save hours of work

Reply:
I hear about the frustration of trying to save hours of work, but I'm not sure about warnings raising an exception - that doesn't happen by default, correct?

Reply:
Correct, it depends on how some random package last set the global warnings filters.

Reply:
I think an exception is unlikely, the warning being suppressed seems more likely. We might want to use a different warning class.

Reply:
@charris which warning class would be better?

Reply:
UserWarning probably (cannot think of a more fitting one), since to me it seemed consensus was formed around making an indefinite warning?

Reply:
UserWarning probably (cannot think of a more fitting one), since to me it seemed consensus was formed around making an indefinite warning? Thanks for picking this up!

Reply:
fixed